src/args.rs

```diff
@@ -76,36 +76,42 @@ fn parse_interface(src: &str) -> Result<IpAddr, std::net::AddrParseError> {
 }
 
 /// Checks wether the auth string is valid, i.e. it follows the syntax username:password
-fn parse_auth(src: &str) -> Result<(String, String), String> {
+fn parse_auth(src: &str) -> Result<auth::RequiredAuth, String> {
     let mut split = src.splitn(2, ':');
+    let errmsg = "Invalid credentials string, expected format is username:password".to_owned();
```

Comment:
@svenstaro I see you used `to_owned()` to convert `&'static str` into `String`. Why didn't you just use `&'static str` directly? _(I asked this because I am learning Rust)_

Reply:
It's because I'm returning this `String` and in order to do that it has to be owned because you can't move &'static str.